lib/hdk_time_indexing/src/sorting.rs

```diff
@@ -0,0 +1,62 @@
+use std::cmp::Ordering;
+use chrono::{DateTime, Utc, NaiveDateTime};
+use hdk::prelude::*;
+use crate::{
+    TimeIndexResult,
+    reading::link_prefix_for_index,
+    index_tree::IndexSegment,
+};
+
+/**
+ * Higher-order function to generate a comparator callback for use in
+ * sorting sets of `EntryHash`es that have been indexed into the
+ * `index_name` time-ordered index.
+ *
+ * It is presumed that the entries are validly linked in the index. If
+ * not, they will be placed at the end of the list in undefined order.
+ *
+ * Usage:
+ *  let mut my_entry_hashes: Vec<EntryHash> = some_list_of_hashes;
+ *  my_entry_hashes.sort_by(sort_entries_by_time_index("my_time_index_name"));
+ */
+pub fn sort_entries_by_time_index<'a, I>(index_name: &'a I) -> Box<dyn for<'r> Fn(&EntryHash, &EntryHash) -> Ordering + 'a>
+    where I: AsRef<str>,
+{
+    let prefix = link_prefix_for_index(index_name);
+
+    Box::new(move |a, b| {
+        let a_timestamp = get_time_for_entry_hash(prefix.to_owned(), a);
+        let b_timestamp = get_time_for_entry_hash(prefix.to_owned(), b);
+        b_timestamp.cmp(&a_timestamp)
+    })
+}
+
+// :TODO: consider an appropriate way of handling errors
+// :TODO: should we adjust for possibility of entries being validly indexed multiple times in the same index?
```

Comment:
I'm a bit behind nowadays on how LinkTag works... can you add a little code comment with context for the index_link_prefix?